### PR TITLE
[New Feature] Fishing Particles Overlay

### PR DESCRIPTION
**Pull request by Phoube**
Yay! It's the feature that literally no one asked for! But I think it's pretty cool so here it is! Lol

I use some terms in the writeup: "fish particles" refer to the particles spawned in the water when a fish is approaching or about to bite on a cast rod. Identifying "your fish particles" means identifying which particles relate to the fish approaching your rod, as opposed to those approaching other player's hooks.

This feature scans through recently spawned fish particles around a cast hook and (with high accuracy) predicts which particle(s) are converging to your cast hook. It then spawns another particle of a different type (I chose a lava drip) on top of the original particle if it's yours. 

TLDR: This is a feature that provides a GUI overlay that tracks how close your fish is to biting on your cast hook. This is super useful for fishing with other people, and especially relevant to the fishing event, during which people pack around a small pond with many other rods/fish particles in the water.

Special consideration was given to computation time, as tracking each particle in the water is not an easy feat. The feature generally takes about 50-100us every tick when the player has a cast rod (well under the 50ms we get every tick).

Please see WorldListener.java for an in-depth explanation of how everything works. And feel free to suggest changes I could make to improve the feature--specifically if you know of a better way of tracking recently spawned particles.

### CODE REVIEW

**Comment by Charzard4261** [comment]
Woah this looks cool

**Comment by Phoube** [comment]
:D

**Comment by Phoube** [comment]
On review I think the tech behind this feature is nice, but the frontend implementation (listener for spawned particles, as well as the GUI of simply spawning in another particle) is messy and could be refined. Let's wait to merge this PR until after a feature refactor.
